src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -268,16 +268,41 @@ public String getTypeDescription() {
       help = "Whether to upload locally executed action results to the remote cache.")
   public boolean remoteUploadLocalResults;
 
+  @Deprecated
   @Option(
       name = "incompatible_remote_build_event_upload_respect_no_cache",
       defaultValue = "false",
       documentationCategory = OptionDocumentationCategory.REMOTE,
       effectTags = {OptionEffectTag.UNKNOWN},
+      deprecationWarning =
+          "--incompatible_remote_build_event_upload_respect_no_cache has been deprecated in favor"
+              + " of --experimental_remote_build_event_upload.",
       help =
           "If set to true, outputs referenced by BEP are not uploaded to remote cache if the"
               + " generating action cannot be cached remotely.")
   public boolean incompatibleRemoteBuildEventUploadRespectNoCache;
 
+  @Option(
+      name = "experimental_remote_build_event_upload",
+      defaultValue = "all",
+      documentationCategory = OptionDocumentationCategory.REMOTE,
+      effectTags = {OptionEffectTag.UNKNOWN},
+      converter = RemoteBuildEventUploadModeConverter.class,
+      help =
+          "If set to 'minimal', local outputs are not uploaded to the remote cache, except for"
+              + " files that are important to the consumers of BES (e.g. test logs and timing"
+              + " profile). Paths for blobs that are already uploaded to the remote cache are"
+              + " converted. Default to 'all'.")
```

Comment:
Does it make sense to mention the paths here? And if so, can the explanation be expanded, since what are they being converted to, and what does it have to do with uploading? (I know the answers to these, but thinking more for the documentation.)

Reply:
Updated doc.